File: server/Models/CentroFilial.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using server.Shared.ComboBox;
using server.Shared.Crudl;

namespace server.Models;

public class CentroFilial : ICrud
{
      [Key]
      public long? Id { get; set; }

      [Required]
      public string Codigo { get; set; }

      [Required]
      public string Nome { get; set; }
      
      [ForeignKey("Empresa")]
      public long? IdEmpresa { get; set; }
    
      public virtual CentroEmpresa? Empresa { get; set; }
      
      public virtual List<ComboBox> Empresas { get; set; } = [];
}

File: client/src/app/centro-filial/centro-filial.ts
import { CentroEmpresa } from "../centro-empresa/centro-empresa";
import { ComboBoxForm } from "../shared/form/combobox-form/combobox-form";

export class CentroFilial {
  Id!: string;
  Codigo!: string;
  Nome!: string;
  IdEmpresa!: string;
  Empresa!: CentroEmpresa;
  Empresas!: ComboBoxForm [];
}